### ISSUE TITLE
[Bug] Strange Behavior of "ws-opts" section

### ISSUE THREAD

**Issue by lostwindsenril** [Dreamacro/clash#1620]
### Verify steps

- [X]  如果你可以自己 debug 并解决的话，提交 PR 吧 Is this something you can **debug and fix**? Send a pull request! Bug fixes and documentation fixes are welcome. 
- [X]  我已经在 [Issue Tracker](……/) 中找过我要提出的问题 I have searched on the [issue tracker](……/) for a related issue. 
- [X]  我已经使用 dev 分支版本测试过，问题依旧存在 I have tested using the dev branch, and the issue still exists. 
- [X]  我已经仔细看过 [Documentation](https://github.com/Dreamacro/clash/wiki/) 并无法自行解决问题 I have read the [documentation](https://github.com/Dreamacro/clash/wiki/) and was unable to solve the issue. 
- [X]  这是 Clash 核心的问题，并非我所使用的 Clash 衍生版本（如 OpenClash、KoolClash 等）的特定问题 This is an issue of the Clash core *per se*, not to the derivatives of Clash, like OpenClash or KoolClash. 

### Clash version

2021.09.07 Premium

### What OS are you seeing the problem on?

Windows

### Clash config

```yaml
proxies:
  - name: example.com
    type: vmess
#    server: example.com
    server: 114.114.114.114
    port: 443
    uuid: *************
    alterId: 0
    cipher: auto
    tls: true
    skip-cert-verify: true
    network: ws
# Bug?
    ws-opts:
      path: /
      headers:
        Host: example.com
#    ws-path: /
#    ws-headers:
#      Host: example.com
```

### Clash log

```shell
time="2021-09-14T14:01:53+08:00" level=warning msg="[TCP] dial PROXY (match RuleSet/Global) to content-firebaseappcheck.googleapis.com:443 error: dial 129.226.13.195:443 error: x509: certificate signed by unknown authority"
time="2021-09-14T14:01:53+08:00" level=warning msg="[TCP] dial PROXY (match RuleSet/Global) to content-firebaseappcheck.googleapis.com:443 error: dial 129.226.13.195:443 error: x509: certificate signed by unknown authority"
time="2021-09-14T14:01:53+08:00" level=warning msg="[TCP] dial PROXY (match Match/) to blog.raveland.org:1313 error: dial 129.226.13.195:443 error: x509: certificate signed by unknown authority"
time="2021-09-14T14:01:53+08:00" level=warning msg="[TCP] dial PROXY (match Match/) to blog.raveland.org:1313 error: dial 129.226.13.195:443 error: x509: certificate signed by unknown authority"
time="2021-09-14T14:01:54+08:00" level=warning msg="[TCP] dial PROXY (match RuleSet/Global) to gitbook-28427.firebaseio.com:443 error: dial 129.226.13.195:443 error: x509: certificate signed by unknown authority"
time="2021-09-14T14:01:54+08:00" level=warning msg="[TCP] dial PROXY (match RuleSet/Global) to gitbook-28427.firebaseio.com:443 error: dial 129.226.13.195:443 error: x509: certificate signed by unknown authority"
```

### Description

Hello.
I read about "ws-path& ws-headers will be deprecated in 2022" from https://lancellc.gitbook.io/clash/clash-config-file/proxies/config-a-vmess-proxy, so I decide to margin into "ws-opts". 
I use IP address instead of domain in "server", and specify Host in "ws-headers". I have a vaild certificate on my server, and the old config worked well.
However, after I added "ws-opts" section, it don't not work any more, even with "skip-cert-verify: true", "servername: example.com" and "tls-hostname: example.com" added (sorry but I can't tell the difference between "servername", "tls-hostname" and "ws-headers"). It can only work when I use domain instead of IP address in "server".
In conclusion, I found "ws-opts" section only accpet domain in "server" when TLS enabled. 
I'm using a fake host with real IP and self signed certificate to cheat "that firewall", since the old  "ws-path& ws-headers" will be deprecated soon, my settings will be also failed in future. Please help me solve this problem.

**Comment by Dreamacro** [comment]
fixed on #1595

**Comment by lostwindsenril** [comment]
> fixed on #1595

Great, and could you please tell me about the difference between "servername", "tls-hostname" and "ws-headers"?  Is it the same if I specify any of them? I use nginx reverse proxy dealing with TLS.

**Comment by Dreamacro** [comment]
I don't know where you are using `tls-hostname`. But it is equivalent to `servername` or `sni`.

header is the http header. When you do not specify a servername and use tls, the header `Host` is used as the servername by default

**Comment by lostwindsenril** [comment]
> I don't know where you are using `tls-hostname`. But it is equivalent to `servername` or `sni`.
> 
> header is the http header. When you do not specify a servername and use tls, the header `Host` is used as the servername by default

Thanks a lot.
